chore(update): add connect dependencies paragraph to upgrade guide (#651)
camunda/camunda-docs-manual: content/update/minor/713-to-714/_index.md

Users that expect a case-sensitive result, will need to adjust their logic, or Task names and descriptions, 
for this change of behavior.

> # Connect Version in Existing Dependencies

Reviewer: Let's try to be more explicit in this section on what people need to do in a particular setup, i.e.:

* If they use embedded engine (including Spring Boot starter) => Inform users about two new dependencies of camunda-engine. Tell that that `connect-core` is required. Tell them that `connectors-all` can be replaced by `connector-dependency`. Explain what the difference is and why they may want to do this. Give an example how that looks in a POM file.
* If they use a shared engine that they update manually => need to add connect modules if they didn't have them before

Users that expect a case-sensitive result, will need to adjust their logic, or Task names and descriptions, 
for this change of behavior.

# New Engine Dependency - Connect

> Camunda Connect dependency has been added to the process engine (`camunda-engine`) artifact. That changes the status of the dependency from optional to required. See below the details:

Reviewer: Let's briefly give the context why this is a new dependency, e.g.: In the context of the new telemetry feature (link to docs), ...

-- In a case of **Embedded engine** scenario (includes **Spring Boot Starter** setups), there are two new dependencies added to the `camunda-engine`:

* `camunda-connect-core` dependency is required from 7.14.0 version and on.
* `camunda-connect-connectors-all` dependency also comes by default. It can be replaced by the `camunda-connect-http-client`. The difference is that `camunda-connect-connectors-all` doesn't have dependencies and contains the HTTP and SOAP connectors, as long as `camunda-connect-http-client` brings a `http-client` dependency. You can see an example below how to exchange the dependencies.

```
<dependency>
  <groupId>org.camunda.bpm</groupId>
  <artifactId>camunda-engine</artifactId>
  <exclusions>
    <exclusion>
      <groupId>org.camunda.connect</groupId>
      <artifactId>camunda-connect-connectors-all</artifactId>
    </exclusion>
  </exclusions>
</dependency>
<dependency>
  <groupId>org.camunda.connect</groupId>
  <artifactId>camunda-connect-http-client</artifactId>
  <scope>runtime</scope>
</dependency>
```

> -- In a case of **Shared engine** scenario, you will need to add the connect modules if they are not present yet to the setup. The respective upgrade guides of the application server installation have been updated accordingly to reflect the change.

Reviewer: ```suggestion
-- In a case of **Shared engine** scenario, you will need to add the connect modules if they are not present yet to the setup. The respective update guides for the application servers contain the necessary steps to do this.
```

> In case you already have a [Connect]({{< ref "/reference/connect/_index.md#maven-coordinates" >}}) dependencies to some of your projects, please consider consolidating the version of them with one that comes as dependency with the engine. That will prevent inconsistencies on the system. Please note that Connect plugin component is still an optional dependency.

Reviewer: ```suggestion
In case you already have a [Connect]({{< ref "/reference/connect/_index.md#maven-coordinates" >}}) dependencies to some of your projects, please consider consolidating the version of them with one that comes as dependency with the engine. That will prevent inconsistencies on the system. Please note that the Connect process engine plugin is still an optional dependency.
```
